src/js/components/Button/StyledButtonKind.js

```diff
@@ -37,7 +37,16 @@ const fontStyle = (props) => {
   `;
 };
 
-const padFromTheme = (size = 'medium', theme, themeObj, kind) => {
+const padFromTheme = (size = 'medium', theme, themeObj, kind, iconOnly) => {
+  if (size && iconOnly && themeObj?.size?.[size]?.iconOnly?.pad) {
+    const pad = themeObj?.size?.[size]?.iconOnly?.pad;
+
+    return {
+      vertical: typeof pad === 'string' ? pad : pad.vertical,
+      horizontal: typeof pad === 'string' ? pad : pad.horizontal,
+    };
```

Comment:
Minor. Not sure if either of these helps readability... 

```suggestion
    return typeof pad === 'object' ? pad : {
      vertical: pad,
      horizontal: pad,
    };
```

OR 

```suggestion
    return {
      vertical: pad.vertical || pad,
      horizontal: pad.horizontal || pad,
    };
```

Reply:
I'm inclined to leave as is.